relay-server/src/envelope.rs

```diff
@@ -954,6 +956,23 @@ impl Envelope {
             .write_all(buf)
             .map_err(EnvelopeError::PayloadIoFailed)
     }
+
+    /// Return the data category type of the event item, if any, in this envelope.
+    pub fn get_event_category(&self) -> Option<DataCategory> {
+        let event_categories: Vec<DataCategory> = self
+            .items()
+            .filter_map(infer_event_category)
+            .unique()
+            .collect();
+        if event_categories.len() == 1 {
+            Some(event_categories[0])
+        } else {
+            if event_categories.len() > 1 {
```

Comment:
Let me know if there is a more graceful way to handle this than `collect()`'ing into a `Vec`. I'm still a newb with Rust's fluent iterators.

Unlike before, I'm more confident that we should never have to print this warning. Tell me if you think there's a better way to handle the condition -- perhaps returning `Some(event_categories[0])` would make more sense.